packages/gasket-plugin-intl/lib/index.js

```diff
@@ -1,102 +1,104 @@
+const workbox = require('./workbox');
+const buildManifest = require('./build-manifest');
+const buildModules = require('./build-modules');
+const { getIntlConfig } = require('./configure');
 
 module.exports = {
   dependencies: ['@gasket/plugin-log'],
   name,
   hooks: {
     init,
-    async initReduxState(gasket, state, req) {
-      const acceptLanguage = (req.headers['accept-language'] || '').split(',')[0];
-      const language = await gasket.execWaterfall('intlLanguage', acceptLanguage, req);
-      return {
-        ...state,
-        intl: {
-          language,
-          assetPrefix: getAssetPrefix(gasket),
-          languageMap: getIntlLanguageMap(gasket),
-          defaultLanguage: getDefaultLanguage(gasket)
-        }
-      };
-    },
-    async create(gasket, { files, pkg }) {
+    configure,
+    async create(gasket, context) {
+      const { files, pkg } = context;
       const rootDir = path.join(__dirname, '..');
       files.add(
-        `${rootDir}/generator/*`,
-        `${rootDir}/generator/**/*`
+        `${ rootDir }/generator/*`,
+        `${ rootDir }/generator/**/*`
       );
 
       pkg.add('dependencies', {
         '@gasket/intl': devDependencies['@gasket/intl'],
         'react-intl': devDependencies['react-intl']
       });
+
+      context.hasGasketIntl = true;
     },
-    build,
-    middleware(gasket) {
-      //
-      // Make the outputDir available from req for loading locale files
-      // during SSR in HOCs
-      //
-      return (req, res, next) => {
-        req.localesDir = getOutputDir(gasket);
-        next();
+    build: {
+      timing: {
+        first: true
+      },
+      handler: async function build(gasket) {
+        const intlConfig = getIntlConfig(gasket);
+        if (intlConfig.modules) {
+          await buildModules(gasket);
+        }
+        await buildManifest(gasket);
+      }
+    },
+    webpack() {
+      // webpack is not listed as a required dependency but if this lifecycle
+      // is invoked we can assume it has been installed by @gasket/plugin-webpack
+      const webpack = require('webpack');
+
+      return {
+        plugins: [
+          new webpack.EnvironmentPlugin([
+            'GASKET_INTL_LOCALES_DIR',
+            'GASKET_INTL_MANIFEST_FILE'
+          ])
+        ]
       };
     },
-    express(gasket, app) {
-      const outputDir = getOutputDir(gasket);
+    middleware(gasket) {
+      const { defaultLocale, basePath, localesMap } = getIntlConfig(gasket);
 
-      app.use('/_locales', serveStatic(outputDir, {
-        index: false,
-        maxAge: '1y',
-        immutable: true
-      }));
+      return async function intlMiddleware(req, res, next) {
+        /* eslint-disable require-atomic-updates */
+        const acceptLanguage = (req.headers['accept-language'] || defaultLocale).split(',')[0];
+        const locale = await gasket.execWaterfall('intlLocale', acceptLanguage, req, res);
```

Comment:
I changed `intlLanguage` to `intlLocale` to standardize our term for this. when moving this lifecycle out of `initReduxState`. I'm wondering if we should still retain `intlLanguage` but with a deprecated warning. Thoughts?

Reply:
I'm okay with removing it since this will be a new major version anyway, but it should be documented clearly (is there a migration document for the new major version?)